server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/MySQLAffectedRows.java

```diff
@@ -0,0 +1,166 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*
+* Copyright (c) 2023 OceanBase.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package com.oceanbase.odc.service.sqlcheck.rule;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.springframework.jdbc.core.JdbcOperations;
+
+import com.oceanbase.odc.core.shared.constant.DialectType;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckContext;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckRule;
+import com.oceanbase.odc.service.sqlcheck.SqlCheckUtil;
+import com.oceanbase.odc.service.sqlcheck.model.CheckViolation;
+import com.oceanbase.odc.service.sqlcheck.model.SqlCheckRuleType;
+import com.oceanbase.tools.sqlparser.statement.Statement;
+import com.oceanbase.tools.sqlparser.statement.delete.Delete;
+import com.oceanbase.tools.sqlparser.statement.insert.Insert;
+import com.oceanbase.tools.sqlparser.statement.update.Update;
+
+import lombok.NonNull;
+
+/**
+ * {@link MySQLAffectedRows}
+ *
+ * @author yiminpeng
+ * @version 1.0
+ * @date 2024-08-01 18:18
+ */
+public class MySQLAffectedRows implements SqlCheckRule {
+
+    private final Integer maxSQLAffectedRows;
+
+    private final JdbcOperations jdbcOperations;
+
+    public MySQLAffectedRows(@NonNull Integer maxSQLAffectedRows, JdbcOperations jdbcOperations) {
+        this.maxSQLAffectedRows = maxSQLAffectedRows <= 0 ? 0 : maxSQLAffectedRows;
+        this.jdbcOperations = jdbcOperations;
+    }
+
+    /**
+     * Get the rule type
+     */
+    @Override
+    public SqlCheckRuleType getType() {
+        return SqlCheckRuleType.RESTRICT_SQL_AFFECTED_ROWS;
+    }
+
+    /**
+     * Execution rule check
+     */
+    @Override
+    public List<CheckViolation> check(@NonNull Statement statement, @NonNull SqlCheckContext context) {
+
+        if (statement instanceof Insert || statement instanceof Update || statement instanceof Delete) {
```

Comment:
What about `Truncate`?

Reply:
It's missing here. I'll make it up later.

Reply:
This issue only supports DML statements and does not involve DDL.

Reply:
`Truncate` belongs to DML. And if it's all about DML, your rule name should be precise. Please double check with @Jane201510

Reply:
Ok, I only paid attention to insert/update/delete during the research.

Reply:
Confirmed with the PM, This issue does not support `Truncate` for the time being, but it will include `select`.

Reply:
Confirmed with the PM, This issue does not support `Truncate` for the time being, but it will include `select`.